system/portenforce.py

```diff
@@ -0,0 +1,182 @@
+---
+module: portenforce
+author:
+    - '"Nathan Davison (@ndavison)" <ndavison85@gmail.com>'
+version_added: "2.1"
+description:
+    - Kill processes listening on TCP and UDP ports not defined in the whitelists.
+short_description: Kill processes listening on TCP and UDP ports not defined in the whitelists.
+options:
+  whitelist_tcp:
+    description:
+      - A list of TCP ports to allow.
+    required: false
+    type: "list"
+  whitelist_udp:
+    description:
+      - A list of UDP ports to allow.
+    required: false
+    type: "list"
+'''
+
+EXAMPLES = '''
+- name: only allow TCP 80, 443, and 22
+  portenforce:
+    whitelist_tcp:
+      - 22
+      - 80
+      - 443
+'''
+
+RETURN = '''
+killed:
+  description: A list of pids that were killed.
+  type: list
+  contains:
+    pid:
+      description: The pid of a killed process.
+      type: int
+      sample: 1223
+    port:
+      description: The port the killed pid was listening on.
+      type: int
+      sample: 443
+    proto:
+      description: The protocol, TCP or UDP, of the listening connection.
+      type: string
+      sample: 'tcp'
+    stime:
+      description: The start time of the killed process.
+      type: string
+      sample: Thu Apr 21 17:30:16 2016
+'''
+
+import re
+from subprocess import Popen, PIPE
+
+def netStatParse(raw):
+    results = list()
+    for line in iter(raw.splitlines()):
+        listening_search = re.search('[^ ]+:[^ ]+', line)
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('[^ ]+:([^ ]+)', splitted[3])
+            pidstr = ''
+            if 'tcp' in splitted[0]:
+                proto = 'tcp'
+                pidstr = splitted[6]
+            elif 'udp' in splitted[0]:
+                proto = 'udp'
+                pidstr = splitted[5]
+            pids = re.search('([0-9]+)/(.*)', pidstr)
+            if conns and pids:
+                port = conns.group(1)
+                pid = pids.group(1)
+                name = pids.group(2)
+                result = dict(pid=int(pid), port=int(port), proto=proto, name=name)
+                if result not in results:
+                    results.append(result)
+            elif not pids:
+                raise EnvironmentError('Could not get the pids for the listening ports - possibly a permission issue')
+    return results
+
+def applyWhitelist(portspids, whitelist=list()):
+    kill_pids = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            if dict(pid=p['pid'], port=p['port'], proto=p['proto'], name=p['name'], stime=p['stime']) not in kill_pids:
+                kill_pids.append(dict(pid=p['pid'], port=p['port'], proto=p['proto'], name=p['name'], stime=p['stime']))
+    return kill_pids
+
+def getPidSTime(pid):
+    p1 = Popen(['ps', '-o', 'lstart', '-p', str(pid)], stdout=PIPE, stderr=PIPE)
+    p2 = Popen(['grep', '^[^ ]'], stdin=p1.stdout, stdout=PIPE, stderr=PIPE)

```

Comment:
I suspect this would take less ressources and be cleaner to do that in python directly.